routers/web/user/setting/secrets.go

```diff
@@ -4,36 +4,24 @@
 package setting
 
 import (
-	"net/http"
-
-	"code.gitea.io/gitea/modules/base"
 	"code.gitea.io/gitea/modules/context"
 	"code.gitea.io/gitea/modules/setting"
 	shared "code.gitea.io/gitea/routers/web/shared/secrets"
 )
 
-const (
-	tplSettingsSecrets base.TplName = "user/settings/secrets"
-)
-
-func Secrets(ctx *context.Context) {
-	ctx.Data["Title"] = ctx.Tr("secrets.secrets")
-	ctx.Data["PageIsSettingsSecrets"] = true
-
+func PrepareSecretsData(ctx *context.Context) {
```

Comment:
There is not only one `PrepareSecretsData`?

Reply:
Yes, this is from different settings (user, organizaition, and repo, but not admin)

Reply:
I mean , the code seems similar? What's the difference?

They just do `shared.SetSecretsContext(ctx,...)`

Reply:
For repo, it has an additional step of setting disablessh `ctx.Data["DisableSSH"] = setting.SSH.Disabled`?

Reply:
`ctx.Data["DisableSSH"] = setting.SSH.Disabled` could be prepared separately.

Reply:
These are actually old codes, should I move them to a shared file? Like under the `web/shared` folder

Reply:
`PrepareSecretsData` functions are merged back into `Secrets` router functions, so might not need to extract the prepare part now.